SYCL/ESIMD/api/functional/functions/functions_atomic_update.hpp

```diff
@@ -0,0 +1,85 @@
+//===-- functions_atomic_update.hpp - Generic code for tests on simd
+//      atomic_update -----------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides common code for tests on simd atomic_update function
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#define ESIMD_TESTS_DISABLE_DEPRECATED_TEST_DESCRIPTION_FOR_LOGS
+
+#include "../common.hpp"
+
+#include <vector>
+
+namespace esimd = sycl::ext::intel::esimd;
+
+namespace esimd_test::api::functional::functions {
+
+// Structure that provides a call operator that lets filter values by different
+// algorithm.
+namespace masks {
+struct ChangeByStep {
+  bool operator()(size_t val) SYCL_ESIMD_KERNEL { return val % 2 == 0; }
+};
+
+struct ChangeAll {
+  bool operator()(size_t val) SYCL_ESIMD_KERNEL { return true; }
+};
+
+struct ChangeNothing {
+  bool operator()(size_t val) SYCL_ESIMD_KERNEL { return false; }
+};
+} // namespace masks
+
+enum class offset { all, ordered_step, non_ordered_step };
```

Comment:
```suggestion
enum class offset_generation { all, ordered_step, non_ordered_step };
```
I've already regretted for too short name suggestion, sorry) Maybe something like this would be less probable to conflict with local variables, how do you think?

Reply:
this comment has been applied [here](https://github.com/intel/llvm-test-suite/pull/939/commits/3a3e37f3d8b5c247a3e92f48f51646d4bf27f524#diff-28ba9c453e6f6e488b2719b183c72b8802b07162628e6fc1445d3fe6d8206cf6R43)